### DISCUSSION TITLE
Filament Export: $options are not propagated to modifyQuery(Builder $query)

### DISCUSSION THREAD

**Discussion by shopauskunft-david** [filamentphp/filament#12794] [Ideas]
Hi,

when using an Filament Exporter to export rows of a table i can pass options:

https://filamentphp.com/docs/3.x/actions/prebuilt-actions/export#using-export-options

i can use this options then afterwards in`formatStateUsing()` and in `limit()` but unfortunately not in an other place where it would be extremely useful - in `modifyQuery()`. Since it is a static method and i can not override it.

Please make it possible to pass options to the query in order for the user to select specific rows via `where()`, for example.

I know that there is a BulkExportAction, but in my use case i have several 10.000 of rows that i have to filter; and nginx is struggling to accept a requests that contains the _specific_ rows to export. This way i need to filter the rows via `options`().

Thanks a lot !

**Comment by DavidKoLab** [comment]
Try the following syntax, injecting $data variable 
```
ExportAction::make()
    ->exporter(ProductExporter::class)
    ->modifyQueryUsing(fn (Builder $query, array $data) => $query->where('is_active', $data['is_active']))
```    
[ Documents_Here](https://filamentphp.com/docs/3.x/actions/advanced)
